test/ChainRulesTests.jl

```diff
@@ -0,0 +1,137 @@
+module ChainRulesTest
+
+using LinearAlgebra
+using ChainRules
+using DiffResults
+using ReverseDiff
+using Test
+
+f(x) = sum(4x .+ 1)
+
+function ChainRules.rrule(::typeof(f), x)
+    r = f(x)
+    function back(d)
+        #=
+        The proper derivative of `f` is 4, but in order to
+        check if `ChainRules.rrule` had taken over the compuation,
+        we define a rrule that returns 3 as `f`'s derivative.
+
+        After importing this rrule into ReverseDiff, if we get 3
+        rather than 4 when we compute the derivative of `f`, it means
+        the importing mechanism works.
+        =#
+        return ChainRules.NoTangent(), fill(3 * d, size(x))
+    end
+    return r, back
+end
+
+ReverseDiff.@grad_from_chainrules f(x::ReverseDiff.TrackedArray)
+
+
+g(x, y) = sum(4x .+ 4y)
+
+function ChainRules.rrule(::typeof(g), x, y)
+    r = g(x, y)
+    function back(d)
+        # same as above, use 3 and 5 as the derivatives
+        return ChainRules.NoTangent(), fill(3 * d, size(x)), fill(5 * d, size(x))
+    end
+    return r, back
+end
+
+ReverseDiff.@grad_from_chainrules g(x::ReverseDiff.TrackedArray, y)
+ReverseDiff.@grad_from_chainrules g(x, y::ReverseDiff.TrackedArray)
+ReverseDiff.@grad_from_chainrules g(x::ReverseDiff.TrackedArray, y::ReverseDiff.TrackedArray)
```

Comment:
It would be good to test that actually after the first `@grad_from_chainrules` call only the desired argument are tracked etc.